src/BSeries.jl

```diff
@@ -1774,6 +1774,34 @@ function energy_preserving_order(rk::RungeKuttaMethod, max_order)
     return p - 1
 end
 
+"""
+    energy_preserving_order(rk::RungeKuttaMethod, max_order, tolerance)
+
+This function checks up to which order a Runge-Kutta method `rk` is
+energy-preserving for Hamiltonian problems, with a parameter of Tolerance. 
+Default tolerance is atol = 0.
+It requires a `max_order` so that it does not run forever if the order up to
+which the method is energy-preserving is too big or infinite.
+
+See also [`is_energy_preserving`](@ref)
+"""
+function energy_preserving_order(rk::RungeKuttaMethod, max_order, tolerance)
```

Comment:
We already have `function energy_preserving_order(rk::RungeKuttaMethod, max_order)` above. Please modify the existing functionality and do not introduce another method.